### DISCUSSION TITLE
What should the Principal be when creating an IAM

### DISCUSSION THREAD

**Discussion by adamvietnam** [artilleryio/artillery#2292] [Artillery]
Hi Guys, 

I'm really enjoying artillery and am now looking to use AWS Lamba for distributed testing.

Following the guide at: https://www.artillery.io/docs/load-testing-at-scale/aws-lambda

I can create the IAM Policy okay but I am stuck on creating the IAM Custom trust policy. 

The default policy as generated by the wizard is:
```javascript
{
	"Version": "2012-10-17",
	"Statement": [
		{
			"Sid": "Statement1",
			"Effect": "Allow",
			"Principal": {},
			"Action": "sts:AssumeRole"
		}
	]
}
``` 

However clicking through to complete the AWS wizard as per the docs gives the error: 

_Missing required field "Principal" cannot be empty_

Any clues to what I should be setting "Principal" to?

**Comment by LiamJBerridge** [comment]
Hey @adamvietnam  

you can set it to a service like this 

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Service": "lambda.amazonaws.com"
            },
            "Action": "sts:AssumeRole"
        }
    ]
}

Hope this helps 🍻

**Comment by adamvietnam** [comment]
Brilliant! That works , thank you.

**Comment by davedbase** [comment]
It would be wonderful if this was added to the documentation. Specifically here: https://www.artillery.io/docs/load-testing-at-scale/aws-fargate#setting-up-the-iam-policy-and-role-from-aws. I was forced by the AWS UI to supply the Principal policy and stumbled on this issue which resolved it. Would be great to help users if it was detailed directly.

**Reply by hassy** [reply]
we're working on simplifying all of this! :) cc @InesNi
